### PR TITLE
Added script to collect mailq metrics

### PR DESCRIPTION
**Pull request by tguentel**

### CODE REVIEW

**Comment by dswarbrick** [comment]
Have you considered using https://github.com/kumina/postfix_exporter (assuming your mail system is postfix)?

**Comment by tguentel** [comment]
Yes, I did. But I dont need so many data. Makes no sense in my case to rollout, run an maintain a separate exporter on all my hosts, when I just want to know, if there is a growing mailq.

File: mailq.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Description: Expose the size of the mail queue from mailq command.
+#
+# Author: Thore Guentel <thore.guentel@tangogolf.de>
+#
+# Results:
+# * 0 and more is the count of the mailq objects
+# * -1 means mailq command is not found
```
[the remark this answers is not in the document]
Comment (hansmi) [reply mailq.sh:9]: What @SuperQ is getting at is that you shouldn't export -1/-2 as metrics values for failures, but instead let the script fail. Something like this:

```
#!/bin/bash

set -e -u -o pipefail

count=$(mailq | grep -c '^[A-F0-9]')

echo '# HELP mailq_size_total Count of e-mails in MTA queue'
echo '# TYPE mailq_size_total gauge'
echo "mailq_size_total=${count}"
```

Note that `set -e` causes the shell to automatically fail if a command fails, e.g. in a variable assignment (try this in an interactive shell: `foo=$(false); echo $?`). [`set -o pipefail` is to get the first non-zero exit code in pipelines](https://www.gnu.org/software/bash/manual/html_node/Pipelines).

File: mailq.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# * -2 means error at checking the mailq
+#
+
+METRIC="mailq_size_total"
+HELPTXT="# HELP $METRIC Count of Mails in the Queue"
+TYPETXT="# TYPE $METRIC gauge"
+
+MAILQCMD=$(which mailq)
+if [ "$MAILQCMD" != "" ]
```
Comment (hansmi) [inline mailq.sh:18]: Just for future reference: [Bash sets the status to 127 for unavailable commands](https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html).

**Review by hansmi — CHANGES_REQUESTED** [review]

**Comment by dswarbrick** [comment]
@tguentel Please address the outstanding suggestions if you still wish to submit this.

**Comment by frittentheke** [comment]
@tguentel  Did you want to close this? Just asking since you did push a new commit just before the close.

**Pull request closed without merging**